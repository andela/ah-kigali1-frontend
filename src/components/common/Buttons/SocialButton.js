import React from "react";
import PropTypes from "prop-types";

const SocialButton = props => {
  const { className, iconName, alt, onClick } = props;
  return (
    <div className={`icon ${className}`} onClick={onClick}>
      <img
        src={require(`../../../assets/icons/${iconName}-icon.svg`)}
        alt={alt}
      />
    </div>
  );
};

SocialButton.propTypes = {
  className: PropTypes.string,
  iconName: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
};
SocialButton.defaultProps = {
  className: ""
};
export default SocialButton;
