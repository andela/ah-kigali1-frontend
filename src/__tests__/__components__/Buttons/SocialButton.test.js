import React from "react";
import { shallow } from "enzyme";
import SocialButton from "../../../components/common/Buttons/SocialButton";

const setUp = props => shallow(<SocialButton {...props} />);
const props = {
  iconName: "fb",
  alt: "fb"
};
describe("SocialButton", () => {
  let component;
  beforeEach(() => {
    component = setUp(props);
  });
  it("should render SocialButton component", () => {
    expect(component).toBeDefined();
  });
});
