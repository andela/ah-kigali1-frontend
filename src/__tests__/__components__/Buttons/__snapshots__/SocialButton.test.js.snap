// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialButton should render SocialButton component 1`] = `
<div
  className="icon "
  onClick={[MockFunction]}
>
  <img
    alt="fb"
    src="test-file-stub"
  />
</div>
`;
